src/components/Popover.tsx

```diff
@@ -113,9 +113,10 @@ export default function Popover({ content, show, children, placement, offset, co
           requires: ['maxSize'],
           fn({ state }) {
             const { width } = state.modifiersData.maxSize
+            const maxWidth = Math.max(width, (boundary?.clientWidth ?? 16) - 16)
             state.styles.popper = {
               ...state.styles.popper,
-              maxWidth: `${width}px`,
+              maxWidth: `${maxWidth}px`,
```

Comment:
why not do width and maxWidth separately?

Reply:
We don't want to set `width` in case the popover can be smaller than the total width - just the `maxWidth`.
I override it here because there were issues in integration where it was being set to 12px, and I couldn't figure out why, but the `boundary` being passed was correct so I just overwrote it with a valid value.

Reply:
Adding a comment inline to explain this.